src/net.cpp

```diff
@@ -1055,6 +1056,24 @@ void CConnman::CreateNodeFromAcceptedSocket(std::unique_ptr<Sock>&& sock,
         m_nodes.push_back(pnode);
     }
 
+    NodeEvictionCandidate candidate{
+        /*id=*/pnode->GetId(),
+        /*m_connected=*/pnode->m_connected,
+        /*m_min_ping_time=*/std::chrono::microseconds::max(),
```

Comment:
I think we need to justify these values for the sake of review.

For example, a default (unset) value for `m_min_ping_time` could reasonably also be 0, so it's not clear in review why it's set to `max()`. For values that have no real meaning until after version handshake, maybe `std::optional` would better communicate their state?

Reply:
I'm gonna try to slim down the `NodeEvictionCandidate` interface a bit because a lot of these values don't need to be set by users of the eviction manager. They could just be default initialized internally in the eviction manager. I'm also gonna add some comments explaining the defaults.

I think `std::optional` would make it a bit to complicated but i'll see maybe it works out nicely.

Reply:
I thought about using `std::optional` here but afaict we would end up treating a `std::nullopt` value for one of these in the same way that we treat them right now w.r.t. to the default values. The default values currently increase the likelihood of eviction for new peers (e.g. we prefer to disconnect peers with slow ping times, so the default is the max possible value) and i have added a comment to document that.